plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -366,7 +366,11 @@ def associate_copy_for_repo(unit, dest_repo, set_content=False):
 
     if set_content:
         new_unit.set_storage_path(os.path.basename(unit._storage_path))
-        new_unit.import_content(unit._storage_path)
+        try:
```

Comment:
i would suggest to put try/block into safe_content_import() same like you did with save_and_import_content(). This way you will avoid to repetitive pieces of code.